protocol/auditlog.go

```diff
@@ -100,6 +116,7 @@ func (l ConiksAuditLog) Insert(addr string, signKey sign.PublicKey,
 	// create the new directory history
 	h = newDirectoryHistory(addr, signKey, snaps[0])
 
+	// FIXME: remove this check --> caller calls Audit() before this function
```

Comment:
If we keep the function signature as-is, the caller has to de-serialize the Response twice (in `Audit` and before calling `Insert`), then wouldn't it make things more complicated? I mean the implementor has to do
```
Audit(response)
deSerialize(response)
Insert(...)
```
You also said that `Audit` will "finally update the snapshots if the checks pass", which means you cannot call `Audit` before this function (actually we cannot call `Audit` before this function since we call `newDirectoryHistory` inside this function). 
I think this correct logic is 
```
Insert() {
  h := newDirectoryHistory()
  h.Audit()
  h.updateVerifiedSTR() //optional if we don't do this in Audit
  log.set(h)
}
```